api/users/views.py

```diff
@@ -45,9 +45,45 @@ def get_user(self, check_permissions=True):
 
 
 class UserList(generics.ListAPIView, ODMFilterMixin):
-    """Users registered on the OSF.
+    """List of users registered on the OSF. *Read-only*.
 
-    You can filter on users by their id, fullname, given_name, middle_name, or family_name.
+    Paginated list of users ordered by the date they registered.  Each resource contains the full representation of the
+    user, meaning a re-fetch is not necessary.
+
+    The subroute [`/me/`](me/) is a special endpoint that always points to the currently logged-in user.
```

Comment:
Opening: https://openscience.atlassian.net/browse/OSF-4640,

But we should support that `me` resolves to the currently logged-in user across the API